src/routes/files/handlers.js

```diff
@@ -111,11 +123,20 @@ export default async function uploadHandler(req, res) {
     }
     try {
       await s3Uploader(buffer, fileName, type);
-      await updateStatus(metadata.id, 'UPLOADED');
+      await updateStatus(metadata.id, fileStatuses.uploaded);
       res.status(200).send({ id: metadata.id });
     } catch (err) {
       if (metadata) {
-        await updateStatus(metadata.id, 'UPLOAD_FAILED');
+        await updateStatus(metadata.id, fileStatuses.uploadFailed);
+      }
+      await handleErrors(req, res, err, logContext);
+    }
+    try {
+      await addToScanQueue({ key: metadata.key });
+      await updateStatus(metadata.id, fileStatuses.queued);
+    } catch (err) {
+      if (metadata) {
+        await updateStatus(metadata.id, fileStatuses.queueingFailed);
       }
       await handleErrors(req, res, err, logContext);
     }
```

Comment:
Where do we set the response if everything works? Do we need to set a body with "success" here?

Reply:
We set the queued status here and the worker then sets an accepted or rejected status depending on the result of the scan.

Reply:
As far as success, the file appears in the file list with a status. If something goes wrong, an error shows up in the ui

Reply:
On line 127 we send back a `200` to the caller of the API, do we need to move that down around line 137 so we only send the caller `200` once the file has successfully been queued?

Reply:
I think the 200 is ok even if queuing fails since the file was uploaded. We might want to have something on the worker that checks to db for files with queuing failed as a status and adds them to the queue. That could be a periodic task.

Reply:
`handleErrors` sets the status of the response so in the case where the file uploads and fails to queue I'm pretty sure we'd send both a `200` response and later a `500`, which I'm sure express will complain about. May be missing something though, so let me know!

Reply:
aha, I forgot about handle errors returning a 500. I think we could just log the error then and not call handle errors.

Reply:
ok, now I am just logging queuing errors and not calling handleErrors

Reply:
> now I am just logging queuing errors

This is fine, but please use the named auditLogger export from logger rather than the default logger for any errors.